### ISSUE TITLE
Question: How to crop from region, keeping the aspect ratio ?

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [bumptech/glide#2898]
Glide allows to do center-crop from a file, as such:

                    Glide.with(this).asBitmap().load(someFile)
                            .apply(RequestOptions.centerCropTransform().skipMemoryCache(true).diskCacheStrategy(DiskCacheStrategy.NONE))
                            .into<SimpleTarget<Bitmap>>(object : SimpleTarget<Bitmap>(width, height) {
                                override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
                                    _bitmap = resource
                                }

                                override fun onLoadFailed(errorDrawable: Drawable?) {
                                    super.onLoadFailed(errorDrawable)
                                }
                            })

That's in case we want to crop from the center.

But what if we want to crop from somewhere else? And from a specific region, while keeping the aspect ratio of this region?

For example, suppose we have 1000x1000 image, and I want to crop a 200x400 part of it. 
For keeping aspect ratio of the cropping part, this means that even if I go outside of the source image, it will move (or resize if really needed) the cropping to make it right.

This means that if for some reason I tell it to crop from 899 in X coordinate, it might actually move the cropping to start from 799, instead of from 899 . 

The sort of protection like of center-crop...

**Comment by AndroidDeveloperLB** [comment]
Maybe I will do the logic of deciding where to crop.
But how do I get the bitmap size (without decoding,using Glide) , and then do the region crop?

**Comment by sjudd** [comment]
Is this a duplicate of #2896?

If you want to use `BitmapRegionDecoder`, you're going to have to write a custom `ResourceDecoder`. You can use `BitmapFactory` to obtain the dimensions of the image. It's hard to follow, but you can take a look at how `Downsampler` works here: https://github.com/bumptech/glide/blob/d939314e04b26709e7e890ba8f88494551111e23/library/src/main/java/com/bumptech/glide/load/resource/bitmap/Downsampler.java#L567.

You can also just write a custom `Transformation`, but that will make scaling challenging if you want to crop out a small part of the image.

**Comment by AndroidDeveloperLB** [comment]
Yes, I had the feeling I've wrote about it somewhere... Sorry.
Yet I know about BitmapRegionDecoder and how to get bitmap size without decoding.
Was just hoping to use this library as it's nicer to use, and was hoping to see something like center-crop, yet for a customized rectangle.

Anyway thank you.
Please do consider adding such a feature, including a way to get the bitmap size before deciding what to do with it.
